vars/pragmasToEnv.groovy

```diff
@@ -7,14 +7,15 @@
  * pragmasToEnv variable
  */
 
-String call(String commit_message) {
+Map call(String commit_message) {
```

Comment:
I'm a little leery of this change.   While it is true that what is being returned is a `Map` are we sure that we are not depending on the cast from `Map` to `String` in the `return`ing of the value?

Reply:
> That method has always been only returning a Map not a String.

This is not necessarily true.  Groovy could have been casting the `Map` to a `String` before returning the value, given that the function was declared to return a `String`.

> Apparently the return statement will return object with out groovy checking to see if the type is compatible with the declaration.

Is this actually documented somewhere or just your assumption based on this not having caused any problems so far?  I would say it's more likely that Groovy is doing the casting and returning the casted `String` value rather than it's returning values without concern of the function's declared return type.

> The usage of "${env.pragmas}" does conversion of the Map to a string.

Again, or is it Groovy doing the casting in the `return`?

Ultimately, it may not matter whether the casting is being done on value before `return` or before assignment of the returned `Map`.  It just makes me leery to do this without knowing who all the callers are and what they are doing with the returned value.

Did any of your test runs with this change in it use an commit pragams?  If so, and if they were processed as expected this is probably safe to do.  Might be worth a quick check to make sure there are no consumers of this function other than code paths that are always exercised.

I'm ultimately just trying to avoid us landing an embarrassing time bomb that goes off some time in our future.

Reply:
The Jenkins groovy scripting console confirms that Groovy is doing a Map to String conversion.  It is using a "{foo=bar}" value instead of the "[foo:bar]" conversion that would be done from string interpolation.

Reverting that change until this can be investigated later, as this is slightly different code than is in the DAOS Jenkinsfile for setting the env.pragmas value.